app/src/main/java/com/lagradost/cloudstream3/ui/download/DownloadAdapter.kt

```diff
@@ -83,108 +66,176 @@ data class DownloadHeaderClickEvent(
 )
 
 class DownloadAdapter(
-    private val clickCallback: (DownloadHeaderClickEvent) -> Unit,
-    private val mediaClickCallback: (DownloadClickEvent) -> Unit,
+    private val onHeaderClickEvent: (DownloadHeaderClickEvent) -> Unit,
+    private val onItemClickEvent: (DownloadClickEvent) -> Unit,
+    private val onItemSelectionChanged: (Int, Boolean) -> Unit,

```

Comment:
You might just want to pass in the viewmodel, but that is up to you.

Reply:
Personally I think keeping the view model logic in fragments keeps things more organized.